### ISSUE TITLE
[RFE] Meshtastic v2 support

### ISSUE THREAD

**Issue by RainerB** [lmatte7/meshtastic-go#16]
I suspect this to be a feature request and not a bug, as it is working with meshtastic firmware 1.2

With meshtastic_go_linux_amd64 (0.1.9) and meshtastic firmware 2.0.10 on a t-beam 1.1 (esp32) I get the following errors:

```
$ ./meshtastic_go_linux_amd64 -p /dev/ttyACM0 info
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x8275287]

goroutine 1 [running]:
main.printPreferences(0x8345b88, 0xa08cb90, 0x0, 0x0, 0x0, 0x0, 0xffa90784)
        /Users/Duke/repos/meshtastic-go/pref.go:61 +0xf7
main.getRadioInfo(0x8345b88, 0xa08cb90, 0x0, 0x0, 0x0, 0x0, 0x82ee2fa)
        /Users/Duke/repos/meshtastic-go/info.go:94 +0x319
main.showAllRadioInfo(0xa08f180, 0x0, 0x0)
        /Users/Duke/repos/meshtastic-go/info.go:37 +0x78
github.com/urfave/cli/v2.(*App).RunAsSubcommand(0xa0aa410, 0xa08f0a0, 0x0, 0x0)
        /Users/Duke/go/pkg/mod/github.com/urfave/cli/v2@v2.3.0/app.go:439 +0x7b5
github.com/urfave/cli/v2.(*Command).startApp(0xa0a2750, 0xa08f0a0, 0x82ee326, 0xffa90791)
        /Users/Duke/go/pkg/mod/github.com/urfave/cli/v2@v2.3.0/command.go:278 +0x4ff
github.com/urfave/cli/v2.(*Command).Run(0xa0a2750, 0xa08f0a0, 0x0, 0x0)
        /Users/Duke/go/pkg/mod/github.com/urfave/cli/v2@v2.3.0/command.go:94 +0x75b
github.com/urfave/cli/v2.(*App).RunContext(0xa0aa340, 0x834601c, 0xa0a6068, 0xa08e100, 0x4, 0x4, 0x0, 0x0)
        /Users/Duke/go/pkg/mod/github.com/urfave/cli/v2@v2.3.0/app.go:313 +0x651
github.com/urfave/cli/v2.(*App).Run(...)
        /Users/Duke/go/pkg/mod/github.com/urfave/cli/v2@v2.3.0/app.go:224
main.Init()
        /Users/Duke/repos/meshtastic-go/cli.go:279 +0x16f2
main.main()
        /Users/Duke/repos/meshtastic-go/main.go:6 +0x17
```
The same scenario with meshtastic firmware 1.2.65 works as expected:
```
$ ./meshtastic_go_linux_amd64 -p /dev/ttyACM0 info

Radio Settings:
Node Number:             ...
GPS:                     true
Number of Channels:      8
Firmware:                1.2.65.0adc5ce
Message Timeout (msec):  300000
Min App Version:         20200

Radio Preferences:
==============================================================================
Field Name                                             | Current Value       |
------------------------------------------------------------------------------
PositionBroadcastSecs                                  | 0                   |
...

```

Please let me know if you need further details. Thanks.

**Comment by lmatte7** [comment]
Hi there! Thanks for opening an issue. Yes this library isn't up to date with Meshtastic 2.0, there are some changes to the protobufs and the channel structure that need to be done before it'll work with the new version. 

I had spoken with some of the folks from the meshtastic project who mentioned they were working on their own Go library, but I don't know if there's been much traction lately. I'll follow up with them and see how things are going. If things have stalled I'll look into updating this project.

**Comment by gibeautc** [comment]
Been over a year, guessing there was no response from them and no real push to update this?

**Comment by lmatte7** [comment]
Yep exactly! I worked with some folks for a bit to put out an official Go library, but as far as I know it's stalled. Upgrading this repo has been on my radar, but I wanted to be sure I wasn't stepping on any toes first. I'll review what all would need to happen to get this repo updated if there's still interest from folks

**Comment by lmatte7** [comment]
\Wanted to post an update that I've started working on this and progress is going along smoothly. I'm hoping to have 2.0 support pushed up sometime in the next few weeks

**Comment by lmatte7** [comment]
I'm happy to report that meshtastic 2.0 functionality has been added as requested! Binaries are available for download on the [release](https://github.com/lmatte7/meshtastic-go/releases/tag/v0.2.0) page, or can be compiled as source. 

I'm going to close this issue as resolved, if there are any bugs with the new release feel free to open a new issue. 

Thank you all for your patience!
